### PR TITLE
Implement basic `find` support for SAP HANA

### PR DESCRIPTION
**Pull request by yonarw**
# Description

Implementation of msg_find for hana handler.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [ ] I ensured that PR title is good enough for the changelog.
- [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [ ] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/hana/hanadb/query.go [hunk 1]
```diff
@@ -14,8 +14,121 @@
...
+	token *resource.Token
+}
+
+// QueryDocuments returns an queryIterator to fetch documents for given SQLParams.
+// If the collection doesn't exist, it returns an empty iterator and no error.
+// If an error occurs, it returns nil and that error, possibly wrapped.
+func (hanaPool *Pool) QueryDocuments(ctx context.Context, qp *QueryParams) (types.DocumentsIterator, error) {
+	// Todo: build correct SQL here
+
+	sqlStmt := fmt.Sprintf("SELECT * FROM %q.%q", qp.DB, qp.Collection)
```
Comment (AlekSi) [inline internal/handlers/hana/hanadb/query.go:54]: `*` could return multiple columns, and that could cause `Scan` below to not work. Let's try to specify a single column name there

Comment (yonarw) [reply internal/handlers/hana/hanadb/query.go:54]: Seems to make no difference, same error 
```go
[msg_find.go:59 hana.(*Handler).MsgFind] [values.go:74 iterator.ConsumeValuesN[...]] [query.go:101 hanadb.(*queryIterator).Next] sql: Scan error on column index 0, name "test": unsupported Scan, storing driver.Value type *protocol.LobOutDescr into type *[]uint8
```
Maybe there is a different variant of Scan or a way to parse the Scan result for only the json?

Comment (AlekSi) [reply internal/handlers/hana/hanadb/query.go:54]: The issue seems to be coming from https://github.com/SAP/go-hdb, so maybe someone there could help

**Comment by yonarw** [comment]
The next obstacle seems to be missing schema information in our query. Does postres include that as default or are there any utility functions to tell sjson.Unmarshal that schema information? I was searching through the postgres handler code but could not figure out how the schema is constructed there ...

**Comment by yonarw** [comment]
Basic querying of documents does work now (without filters). We need to clean the code a bit and we still get one warning on queries:
```go
WARN    // 172.18.0.3:60188 -> 172.17.0.1:27017         clientconn/conn.go:605  Proxy response header: length:   111, id:   32, response_to:   19, opcode: OP_MSG
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).logResponse
        /home/jonathan/dev/FerretDB/internal/clientconn/conn.go:605
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run
        /home/jonathan/dev/FerretDB/internal/clientconn/conn.go:318
github.com/FerretDB/FerretDB/internal/clientconn.acceptLoop.func1
        /home/jonathan/dev/FerretDB/internal/clientconn/listener.go:321
2023-09-13T16:04:32.762+0200    WARN    // 172.18.0.3:60188 -> 172.17.0.1:27017         clientconn/conn.go:606  Proxy response message:
{
  "Checksum": 0,
  "FlagBits": 0,
  "Sections": [
    {
      "Document": {
        "$k": [
          "ok",
          "errmsg",
          "code",
          "codeName"
        ],
        "ok": {
          "$f": 0
        },
        "errmsg": "cursor id 1 not found",
        "code": 43,
        "codeName": "CursorNotFound"
      },
      "Kind": 0
    }
  ]
}
```
I guess we are not using the correct way to get the cursorID in [msg_find.go:91](https://github.com/yonarw/FerretDB/blob/5b35a7650f699864707c8725c03c5a56874619f2/internal/handlers/hana/msg_find.go?plain=1L91)?

**Pull request closed without merging**